### PR TITLE
feat: remove patch on cart when changing price mode

### PR DESCRIPTION
**Pull request by nsilva-spryker**
Requirements have change regarding price mode patch update. Right now **we will not patch cart with price mode**

### CODE REVIEW

File: libs/domain/cart/src/services/adapter/default-cart.adapter.ts [hunk 1]
```diff
@@ -79,44 +78,6 @@ export class DefaultCartAdapter implements CartAdapter {
     );
   }
 
-  update(data: UpdateCartQualifier): Observable<Cart> {
```
Comment (supproduction) [inline libs/domain/cart/src/services/adapter/default-cart.adapter.ts]: I would keep update method, just delete headers

File: libs/domain/cart/src/services/default-cart.service.ts [hunk 2]
```diff
@@ -112,28 +111,6 @@ export class DefaultCartService implements CartService {
     },
   ]);
 
-  protected updatePriceMode$ = createEffect<Cart>([
```
Comment (supproduction) [inline libs/domain/cart/src/services/default-cart.service.ts:115]: Let's create update method in service

**Review by supproduction — CHANGES_REQUESTED** [review]

File: libs/domain/cart/src/services/cart.providers.ts [hunk 3]
```diff
@@ -29,10 +27,6 @@ export const cartProviders: Provider[] = [
     provide: TotalsService,
     useClass: DefaultTotalsService,
   },
-  {
-    provide: PriceModeChangeGuard,
-    useClass: CartPriceModeChangeGuard,
-  },
```
Comment (supproduction) [inline libs/domain/cart/src/services/cart.providers.ts:35]: Also, please delete interceptor from public API? (means export)

**Review by supproduction — CHANGES_REQUESTED** [review]

File: libs/domain/cart/src/services/default-cart.service.ts [hunk 4]
```diff
@@ -112,27 +111,24 @@ export class DefaultCartService implements CartService {
...
-              });
-            })
-          )
-        );
-      }
+  update(data: Cart): Observable<Cart | undefined> {
+    if (data) {
+      subscribeReplay(
+        this.getCart().pipe(
+          take(1),
```
Comment (supproduction) [inline libs/domain/cart/src/services/default-cart.service.ts]: Please, use similar flow as `addEntry...`

**Review by supproduction — CHANGES_REQUESTED** [review]

File: libs/domain/cart/src/services/cart.providers.ts (hunk 3 above)
Comment (nsilva-spryker) [reply libs/domain/cart/src/services/cart.providers.ts:35]: `export const PriceModeChangeGuard = 'oryx.PriceModeChangeGuard*'`
Was deleted from file `price-mode.service.ts`

Also CartVersionInterceptor deleted

**Review by supproduction — APPROVED** [review]

**Pull request merged**